test/functional/unified-spec-runner/entities.ts

```diff
@@ -102,12 +104,21 @@ export class UnifiedMongoClient extends MongoClient {
     connectionCheckedInEvent: 'connectionCheckedIn'
   } as const;
 
+  static connectionString(uri: string, uriOptions = {}): string {
+    const connectionString = new ConnectionString(uri);
+    for (const [name, value] of Object.entries(uriOptions ?? {})) {
+      connectionString.searchParams.set(name, String(value));
+    }
+    return connectionString.toString();
+  }
+
   constructor(url: string, description: ClientEntity) {
-    super(url, {
+    super(UnifiedMongoClient.connectionString(url, description.uriOptions), {
       monitorCommands: true,
       ...description.uriOptions,
```

Comment:
does it actually make sense to still pass in the uriOptions separately?

Reply:
does the unified runner spec have a concept of non-uri client options?

Reply:
Oop no that's an over sight we should just have the options in the uri now. There's nothing in the schema for unified runner for general options via the second param.